cli/pkg/stack_mgr/stack_service.go

```diff
@@ -356,6 +319,57 @@ func (s *StackService) GetStacks(ctx context.Context) (map[string]*Stack, error)
 	return s.stacks, nil
 }
 
+func (s *StackService) CollectStackInfo(ctx context.Context, listAll bool, app string) ([]StackInfo, error) {
+	g, ctx := errgroup.WithContext(ctx)
+	stacks, err := s.GetStacks(ctx)
+	if err != nil {
+		return nil, err
+	}
+	// Iterate in order
+	stackNames := maps.Keys(stacks)
+	stackInfos := make([]*StackInfo, len(stackNames))
+	sort.Strings(stackNames)
+	for i, name := range stackNames {
+		i, name := i, name // https://golang.org/doc/faq#closures_and_goroutines
```

Comment:
Nice!

Reply:
ya I noticed when there was a lot of stacks, it was taking a uncomfortably long time to make the list. This should help. I think we have a lot of improvements to make regarding concurrency.